src/error.rs

```diff
@@ -0,0 +1,11 @@
+use thiserror::Error;
```

Comment:
I think we should add the apache license header for all files.

Reply:
We can use `license-eye header fix` to fix all errors about licenses.